synapse/tests/test_lib_storm.py

```diff
@@ -2826,6 +2826,77 @@ async def get(self, name):
             ]
             self.eq(exp, evnts)
 
+    async def test_storm_pkg_onload_active(self):
+        pkg = {
+            'name': 'testload',
+            'version': '0.3.0',
+            'modules': (
+                {
+                    'name': 'testload',
+                    'storm': 'function x() { return((0)) }',
+                },
+            ),
+            'onload': '''
+                $lib.print(testprint)
+                $lib.warn(testwarn)
+
+                $vers = $lib.globals.get(testload:version, (0))
+                $vers = ($vers + 1)
+                $lib.globals.set(testload:version, $vers)
+            '''
+        }
+
+        with self.getTestDir() as dirn:
+            dirn00 = s_common.gendir(dirn, 'core00')
+            dirn01 = s_common.gendir(dirn, 'core01')
+
+            async with self.getTestCore(dirn=dirn00) as core00:
+
+                waiter = core00.waiter(2, 'core:pkg:onload:start', 'core:pkg:onload:complete')
+
+                await core00.addStormPkg(pkg)
+
+                events = await waiter.wait(timeout=10)
+                self.eq(events, [
+                    ('core:pkg:onload:start', {'pkg': 'testload'}),
+                    ('core:pkg:onload:complete', {'pkg': 'testload'}),
+                ])
+
+                self.eq(await core00.callStorm('return($lib.globals.get(testload:version))'), 1)
+
+            s_tools_backup.backup(dirn00, dirn01)
+
+            async with self.getTestCore(dirn=dirn00) as core00:
+                waiter = core00.waiter(1, 'core:pkg:onload:complete')
```

Comment:
I like having the observable events for testing; but this waiter feels like we could be running into a future race condition if the coro finishes during `anit` and before we await the waiter.

Could the onload event storm code push to a Cortex queue? We can wait for items there in a non-culling fashion?

Reply:
Do we have to account for the # of events being fired during replay tests as well?